### PR TITLE
[Data] [Docs] Doc updates

### PR DESCRIPTION
**Pull request by amogkam**
- Adds comments to object detection example `map_batches`
- Adds stats section to Inspecting Data
- Clarifies that transformations are lazy

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by raulchen — APPROVED** [review]

File: doc/source/data/inspecting-data.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ This guide shows you how to:
 * `Describe datasets <#describing-datasets>`_
 * `Inspect rows <#inspecting-rows>`_
 * `Inspect batches <#inspecting-batches>`_
+* `Inspect Dataset runtime statistics <#inspecting-stats>`_
```
Comment (bveeramani) [inline doc/source/data/inspecting-data.rst:14]:
```suggestion
* `Inspect runtime statistics <#inspecting-stats>`_
```

Do we need "Dataset" here? "Inspect Dataset runtime statistics" is more verbose than the other bullet points

File: doc/source/data/inspecting-data.rst [hunk 2]
```diff
@@ -143,3 +144,53 @@ of the returned batch, set ``batch_format``.
 For more information on working with batches, see
 :ref:`Transforming batches <transforming_batches>` and
 :ref:`Iterating over batches <iterating-over-batches>`.
+
+
+Inspecting runtime statistics
+=============================
+
+Ray Data calculates statistics during execution like runtime and memory usage for the different stages.
```
Comment (bveeramani) [inline doc/source/data/inspecting-data.rst:152]: What is "runtime" in this context? Does it mean latency?

File: doc/source/data/inspecting-data.rst [hunk 3]
```diff
@@ -143,3 +144,53 @@ of the returned batch, set ``batch_format``.
...
 :ref:`Transforming batches <transforming_batches>` and
 :ref:`Iterating over batches <iterating-over-batches>`.
+
+
+Inspecting runtime statistics
+=============================
+
+Ray Data calculates statistics during execution like runtime and memory usage for the different stages. 
+
+View debug stats for your Dataset executions via :meth:`Dataset.stats() <ray.data.Dataset.stats>` on an executed dataset. The stats are also persisted under `/tmp/ray/session_*/logs/ray-data.log`.
```
Comment (bveeramani) [inline doc/source/data/inspecting-data.rst:154]:
```suggestion
To view debug stats about your `:class:<~ray.data.dataset.Dataset>` executions, call :meth:`Dataset.stats() <ray.data.Dataset.stats>` on an executed dataset. The stats are also persisted under `/tmp/ray/session_*/logs/ray-data.log`.
```

image

Style guide says don't use via.

This is the first reference to `Dataset` in this section, so let's cross-reference it?

Comment (bveeramani) [inline doc/source/data/inspecting-data.rst:154]: What does "Dataset executions" mean? Does it mean transformations and reads and writes?

File: doc/source/data/inspecting-data.rst [hunk 4]
```diff
@@ -143,3 +144,53 @@ of the returned batch, set ``batch_format``.
...
+    import ray
+    import time
+
+    def pause(x):
+        time.sleep(.0001)
+        return x
+
+    ds = ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
+    ds = ds.map(lambda x: x)
+    ds = ds.map(pause)
```
Comment (bveeramani) [inline doc/source/data/inspecting-data.rst:166]: For consistency with the other code snippets in the user guides, let's chain the method calls?

(Not sure if the syntax in the suggestion is totally correct)
```suggestion
    ds = (
        ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
        .map(lambda x: x)
        .map(pause)
    )
```

**Review by bveeramani — CHANGES_REQUESTED** [review]

File: doc/source/data/inspecting-data.rst (hunk 1 above)
Comment (amogkam) [reply doc/source/data/inspecting-data.rst]: updated

File: doc/source/data/inspecting-data.rst (hunk 2 above)
Comment (amogkam) [reply doc/source/data/inspecting-data.rst:152]: updated to execution

File: doc/source/data/inspecting-data.rst (hunk 3 above)
Comment (amogkam) [reply doc/source/data/inspecting-data.rst]: updated

File: doc/source/data/transforming-data.rst [hunk 5]
```diff
@@ -7,13 +7,16 @@ Transforming Data
...
+    Transformations are lazy by default. They aren't executed until you trigger consumption of the data by :ref:`iterating over the Dataset <iterating-over-data>`, :ref:`saving the Dataset <saving-data>`, or :ref:`inspecting properties of the Dataset <inspecting-data>`.
+
 This guide shows you how to:
 
 * `Transform rows <#transforming-rows>`_
 * `Transform batches <#transforming-batches>`_
 * `Groupby and transform groups <#groupby-and-transforming-groups>`_
 * `Shuffle rows <#shuffling-rows>`_
-* `Repartition data <#repartitioning-data>`_
+* `Repartition data <#repartitioning-data>`_\
```
Comment (bveeramani) [inline doc/source/data/transforming-data.rst:19]: Is this supposed to be here?
```suggestion
* `Repartition data <#repartitioning-data>`_
```

File: doc/source/data/inspecting-data.rst (hunk 2 above)
Comment (amogkam) [reply doc/source/data/inspecting-data.rst:152]: ok updated to wall clock time to make it more precise

**Review by bveeramani — APPROVED** [review]

**Review by scottjlee — APPROVED** [review]

**Pull request merged**